x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/vector/Knn.java

```diff
@@ -173,14 +191,24 @@ private TypeResolution resolveQuery() {
         );
     }
 
+    private TypeResolution resolveK() {
+        if (k == null) {
+            // Function has already been rewritten and included in QueryBuilder - otherwise parsing would have failed
+            return TypeResolution.TYPE_RESOLVED;
+        }
+
+        return isType(k(), dt -> dt == INTEGER, sourceText(), THIRD, "integer").and(isFoldable(k(), sourceText(), THIRD))
```

Comment:
are we missing a test here on what would happen if `k` is not a constant (or not foldable)?